### PR TITLE
fix: binary install was broken for linux

### LINKED ISSUE #718: Latest binary version (4.2.0) doesn't work in MacOS
**Describe the bug**

Latest binary version (4.20) doesn't work in MacOS 10.14.6 

**To Reproduce**

```
spectral ~/Downloads/spectral-macos lint openapi.yml
Error: @stoplight/path: Cannot parse /Users/nogates/Downloads/spectral-macos/openapi.yml because it is not a string
    at parse (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:6984)
    at Object.isAbsolute (/snapshot/spectral/node_modules/@stoplight/path/index.cjs.js:1:7876)
    at Object.<anonymous> (/snapshot/spectral/dist/cli/services/linter.js:40:65)
    at Generator.next (<anonymous>)
    at /snapshot/spectral/node_modules/tslib/tslib.js:110:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/snapshot/spectral/node_modules/tslib/tslib.js:106:16)
    at Object.lint (/snapshot/spectral/dist/cli/services/linter.js:36:20)
    at Object.handler (/snapshot/spectral/dist/cli/commands/lint.js:86:25)
    at Object.runCommand (/snapshot/spectral/node_modules/yargs/lib/command.js:235:26)
```
Also, version is not reported correctly

```
spectral ~/Downloads/spectral-macos --version
0.0.0
```

Previous version (4.1.1) works just fine

Tried both, using the script to pull the latest binary and manually downloaded it via GH releases

### PR DESCRIPTION
**Pull request by philsturgeon**
Related to #718

**Checklist**

- [ ] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Screenshots**

**Additional context**

Add any other context about the pull request here. Remove this section if there is no additional context.

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -11,20 +11,20 @@ if [ "$UNAME" != "Linux" ] && [ "$UNAME" != "Darwin" ] ; then
 fi
 
 if [ "$UNAME" = "Darwin" ] ; then
-  PLATFORM="macos"
+  FILENAME="spectral-macos"
 elif [ "$UNAME" = "Linux" ] ; then
-  PLATFORM="linux"
```
Comment (philsturgeon) [inline scripts/install.sh]: Right now we have `spectral-macos` for macos and `spectral` for linux. This should be fixed properly another time but quick fix here is to point to the right file.

File: scripts/install.sh [hunk 2]
```diff
@@ -11,20 +11,20 @@ if [ "$UNAME" != "Linux" ] && [ "$UNAME" != "Darwin" ] ; then
...
+URL="https://github.com/stoplightio/spectral/releases/latest/download/${FILENAME}"
+SRC="$(pwd)/${FILENAME}"
 DEST=/usr/local/bin/spectral
 
 STATUS=$(curl -sL -w %{http_code} -o $SRC $URL)
 if [ $STATUS -ge 200 ] & [ $STATUS -le 308 ]; then
   mv $SRC $DEST
   chmod +x $DEST
-  echo "Spectral installation was successful"
+  echo "Spectral was installed to: ${DEST}"
```
Comment (philsturgeon) [inline scripts/install.sh]: This is just a little developer experience thing. If you put something on my computer, tell me where. :)

Comment (XVincentX) [reply scripts/install.sh]: Very good point, we might want to back port this in Prism also.

File: src/cli/services/__tests__/linter.test.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { join, resolve } from '@stoplight/path';
+import { join, resolve } from 'path';
```
Comment (XVincentX) [inline src/cli/services/__tests__/linter.test.ts]: Are these changes intended? Now tests are failing 🤔

**Pull request merged**